flang/lib/Lower/Bridge.cpp

```diff
@@ -1473,7 +1473,25 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   /// the global to the ModuleOp as a new uniqued symbol and initialize it with
   /// the correct value. It will be referenced on demand using `fir.addr_of`.
   void instantiateGlobal(const Fortran::lower::pft::Variable &var) {
-    llvm_unreachable("Varun: put your code here");
+    const auto &sym = var.getSymbol();
+    std::string globalName = uniqueCGIdent(sym.name().ToString());
+    auto global = builder->getNamedGlobal(globalName);
+    if (!global) {
```

Comment:
combine these two lines like
```
if (builder->getNamedGlobal(globalName))
  return;
```
This eliminates the indentation of most of the code.